README.md

```diff
@@ -1 +1,63 @@
-# angular-what-is-a-custom-directive-readme
\ No newline at end of file
+# What is a custom Directive?
+
+## Overview
+
+One of the massive powerful aspects of Angular is the ability to actually be able to create our own directives.
+
+## Objectives
+
+- Describe what components are
+- Describe custom Directives
+- Create a custom Directive
+
+## Components
+
+Nowadays, most frontend applications are made up of small, reusable components. If we imagine a car, it would be made up with components such as wheels, windows, engine, etc.
+
+We can create these components using directives. We've already used directives that Angular gives us, but we can also make our own.
+
+Directives can both be HTML attributes and HTML elements. Most of our directives will be HTML elements, as they will have their own template inside of themselves instead of modifying the behaviour of existing elements.
+
+If we imagine an `ng-repeat`, that modifies the behaviour of the element it is placed on by repeating it for every item in an array.
+
+However, we might create a custom directive called "dropdown". This dropdown directive will have it's own template that will get put into the DOM, so we create it as a HTML element instead.
```

Comment:
Here, without actually showing how to create a directive, show an example of what a dropdown menu might look like without using a directive, and then what inserting it would look like with a directive. This will show the reader how much cleaner it is to use a directive.